### DISCUSSION TITLE
Segmentation Fault after running ros node.

### DISCUSSION THREAD

**Discussion by Ajit-M** [OpenVSLAM-Community/openvslam_ros#105] [Q&A] [answered]
Hello, first of all, thank you for making this library open-source. I have been trying to get this package running on the Noetic, but as soon as I run the node it throws a segmentation fault error, and the node crashes. Any ideas about what is causing the segmentation fault error?

![error](https://user-images.githubusercontent.com/20818787/198575722-e717893c-17be-42f0-98a1-929f0c004e87.png)

**Comment by ymd-stella** [comment]
How did you build it?

**Comment by Ajit-M** [comment]
The stella_vslam library I built it with these flags,

cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr/local \
    -DCMAKE_CXX_FLAGS=-std=c++14 \
    -DBUILD_SHARED_LIBS=ON \
    -DBUILD_UNITTESTS=OFF \
    -DG2O_USE_CHOLMOD=OFF \
    -DG2O_USE_CSPARSE=ON \
    -DG2O_USE_OPENGL=ON \
    -DG2O_USE_OPENMP=OFF \
    -DBUILD_WITH_MARCH_NATIVE=OFF \
    ..

I have seen few issues with the g2o library as well, so for g20 I built it using these flags,

cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr/local \
    -DCMAKE_CXX_FLAGS=-std=c++11 \
    -DBUILD_SHARED_LIBS=ON \
    -DBUILD_UNITTESTS=OFF \
    -DG2O_USE_CHOLMOD=OFF \
    -DG2O_USE_CSPARSE=ON \
    -DG2O_USE_OPENGL=ON \
    -DG2O_USE_OPENMP=ON \
    ..

For the stella_vslam_ros, I used these flags

catkin_make -DUSE_PANGOLIN_VIEWER=ON -DUSE_SOCKET_PUBLISHER=OFF

**Comment by ymd-stella** [comment]
Try building stella_vslam with `-DCMAKE_BUILD_TYPE=RelWithDebInfo` instead of `-DCMAKE_BUILD_TYPE=Release` and run it again.

**Comment by ymd-stella** [comment]
In addition, add `-DUSE_STACK_TRACE_LOGGER=ON` to the catkin_make options.

**Comment by Ajit-M** [comment]
I built stella_vslam again by changing the flags you mentioned, here is the log

**(One thing I just noticed is the conflict of OpenCV header files, could that be the reason?)**

luffy@roger:~/hydra_slam/stella_vslam/build$ make -j6
[  0%] Building CXX object googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[  1%] Building CXX object test/helper/CMakeFiles/test_helper.dir/bearing_vector.cc.o
[  1%] Building CXX object test/helper/CMakeFiles/test_helper.dir/landmark.cc.o
[  2%] Building CXX object test/helper/CMakeFiles/test_helper.dir/keypoint.cc.o
[  3%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/config.cc.o
[  3%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/system.cc.o
[  4%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/tracking_module.cc.o
[  5%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/mapping_module.cc.o
[  6%] Linking CXX shared library libtest_helper.so
[  6%] Built target test_helper
[  7%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/global_optimization_module.cc.o
[  7%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/camera/base.cc.o
[  8%] Linking CXX shared library ../../lib/libgtest.so
[  8%] Built target gtest
[  9%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/camera/perspective.cc.o
[ 10%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/camera/fisheye.cc.o
[ 11%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/camera/equirectangular.cc.o
[ 12%] Building CXX object googletest-build/googletest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[ 13%] Linking CXX shared library ../../lib/libgtest_main.so
[ 13%] Built target gtest_main
[ 13%] Building CXX object googletest-build/googlemock/CMakeFiles/gmock.dir/src/gmock-all.cc.o
[ 14%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/camera/radial_division.cc.o
[ 14%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/bow_vocabulary.cc.o
[ 15%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/common.cc.o
[ 16%] Linking CXX shared library ../../lib/libgmock.so
[ 16%] Built target gmock
[ 17%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/frame.cc.o
[ 18%] Building CXX object googletest-build/googlemock/CMakeFiles/gmock_main.dir/src/gmock_main.cc.o
[ 19%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/keyframe.cc.o
[ 20%] Linking CXX shared library ../../lib/libgmock_main.so
[ 20%] Built target gmock_main
[ 20%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/landmark.cc.o
[ 21%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/marker.cc.o
[ 22%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/marker2d.cc.o
[ 23%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/graph_node.cc.o
[ 23%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/camera_database.cc.o
[ 24%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/orb_params_database.cc.o
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/keyframe.cc: In static member function ‘static std::shared_ptr<stella_vslam::data::keyframe> stella_vslam::data::keyframe::from_stmt(sqlite3_stmt*, stella_vslam::data::camera_database*, stella_vslam::data::orb_params_database*, stella_vslam::data::bow_vocabulary*, unsigned int)’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/keyframe.cc:107:79: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘class cv::KeyPoint’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
  107 |     std::memcpy(undist_keypts.data(), p, sqlite3_column_bytes(stmt, column_id));
      |                                                                               ^
In file included from /home/luffy/hydra_slam/stella_vslam/src/stella_vslam/type.h:13,
                 from /home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/common.h:4,
                 from /home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/keyframe.cc:1:
/usr/local/include/opencv2/core/types.hpp:699:27: note: ‘class cv::KeyPoint’ declared here
  699 | class CV_EXPORTS_W_SIMPLE KeyPoint
      |                           ^~~~~~~~
[ 25%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/map_database.cc.o
[ 26%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/bow_database.cc.o
[ 27%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/data/frame_statistics.cc.o
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc: In member function ‘bool stella_vslam::data::map_database::save_keyframes_to_db(sqlite3*, const string&) const’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc:678:9: warning: unused variable ‘ret’ [-Wunused-variable]
  678 |     int ret = SQLITE_ERROR;
      |         ^~~
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc: In member function ‘bool stella_vslam::data::map_database::save_landmarks_to_db(sqlite3*, const string&) const’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc:704:9: warning: unused variable ‘ret’ [-Wunused-variable]
  704 |     int ret = SQLITE_ERROR;
      |         ^~~
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc: In member function ‘bool stella_vslam::data::map_database::save_associations_to_db(sqlite3*, const string&) const’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/data/map_database.cc:785:9: warning: unused variable ‘ret’ [-Wunused-variable]
  785 |     int ret = SQLITE_ERROR;
      |         ^~~
[ 27%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/feature/orb_params.cc.o
[ 28%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/feature/orb_extractor.cc.o
[ 29%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/feature/orb_extractor_node.cc.o
[ 30%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/feature/orb_impl.cc.o
[ 30%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/initialize/base.cc.o
[ 31%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/initialize/perspective.cc.o
[ 32%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/initialize/bearing_vector.cc.o
[ 33%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/io/trajectory_io.cc.o
[ 33%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/io/map_database_io_msgpack.cc.o
[ 34%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/io/map_database_io_sqlite3.cc.o
[ 35%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/marker_detector/base.cc.o
[ 36%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/marker_detector/aruco_disabled.cc.o
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/io/map_database_io_sqlite3.cc: In member function ‘virtual void stella_vslam::io::map_database_io_sqlite3::save(const string&, const stella_vslam::data::camera_database*, const stella_vslam::data::orb_params_database*, const stella_vslam::data::map_database*)’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/io/map_database_io_sqlite3.cc:20:75: warning: unused parameter ‘orb_params_db’ [-Wunused-parameter]
   20 |                                    const data::orb_params_database* const orb_params_db,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
[ 37%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/marker_model/aruco.cc.o
[ 37%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/marker_model/base.cc.o
[ 38%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/area.cc.o
[ 39%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/bow_tree.cc.o
[ 40%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/fuse.cc.o
[ 40%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/projection.cc.o
[ 41%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/robust.cc.o
[ 42%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/match/stereo.cc.o
[ 43%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/initializer.cc.o
[ 43%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/relocalizer.cc.o
[ 44%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/frame_tracker.cc.o
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/module/relocalizer.cc: In member function ‘bool stella_vslam::module::relocalizer::refine_pose_by_local_map(stella_vslam::data::frame&, const std::shared_ptr<stella_vslam::data::keyframe>&) const’:
/home/luffy/hydra_slam/stella_vslam/src/stella_vslam/module/relocalizer.cc:301:14: warning: unused variable ‘num_additional_matches’ [-Wunused-variable]
  301 |         auto num_additional_matches = projection_matcher.match_frame_and_landmarks(curr_frm, local_landmarks, lm_to_reproj, lm_to_x_right, lm_to_scale, margin);
      |              ^~~~~~~~~~~~~~~~~~~~~~
[ 45%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/keyframe_inserter.cc.o
[ 46%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/two_view_triangulator.cc.o
[ 47%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/local_map_cleaner.cc.o
[ 47%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/local_map_updater.cc.o
[ 48%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/loop_detector.cc.o
[ 49%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/module/loop_bundle_adjuster.cc.o
[ 50%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/pose_optimizer.cc.o
[ 50%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/local_bundle_adjuster.cc.o
[ 51%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/transform_optimizer.cc.o
[ 52%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/graph_optimizer.cc.o
[ 53%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/global_bundle_adjuster.cc.o
[ 53%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/optimize/terminate_action.cc.o
[ 54%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/publish/frame_publisher.cc.o
[ 55%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/publish/map_publisher.cc.o
[ 56%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/solve/common.cc.o
[ 57%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/solve/homography_solver.cc.o
[ 57%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/solve/fundamental_solver.cc.o
[ 58%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/solve/essential_solver.cc.o
[ 59%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/solve/pnp_solver.cc.o
[ 60%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/angle.cc.o
[ 60%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/converter.cc.o
[ 61%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/image_converter.cc.o
[ 62%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/random_array.cc.o
[ 63%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/sqlite3.cc.o
[ 63%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/stereo_rectifier.cc.o
[ 64%] Building CXX object src/stella_vslam/CMakeFiles/stella_vslam.dir/util/yaml.cc.o
[ 65%] Linking CXX shared library ../../lib/libstella_vslam.so
[ 65%] Built target stella_vslam
[ 65%] Building CXX object test/CMakeFiles/test_stella_vslam_feature_orb_params.dir/stella_vslam/feature/orb_params.cc.o
[ 66%] Building CXX object test/CMakeFiles/test_stella_vslam_data_bow_vocabulary.dir/stella_vslam/data/bow_vocabulary.cc.o
[ 67%] Building CXX object test/CMakeFiles/test_stella_vslam_match_base.dir/stella_vslam/match/base.cc.o
[ 68%] Building CXX object test/CMakeFiles/test_stella_vslam_data_common_get_cell_indices.dir/stella_vslam/data/common_get_cell_indices.cc.o
[ 68%] Building CXX object test/CMakeFiles/test_stella_vslam_solve_essential_solver.dir/stella_vslam/solve/essential_solver.cc.o
[ 68%] Building CXX object src/pangolin_viewer/CMakeFiles/pangolin_viewer.dir/viewer.cc.o
[ 69%] Linking CXX executable test_stella_vslam_feature_orb_params
[ 70%] Linking CXX executable test_stella_vslam_match_base
[ 70%] Built target test_stella_vslam_match_base
[ 70%] Built target test_stella_vslam_feature_orb_params
[ 71%] Building CXX object test/CMakeFiles/test_stella_vslam_feature_orb_extractor.dir/stella_vslam/feature/orb_extractor.cc.o
[ 72%] Building CXX object test/CMakeFiles/test_stella_vslam_solve_fundamental_solver.dir/stella_vslam/solve/fundamental_solver.cc.o
[ 73%] Linking CXX executable test_stella_vslam_data_bow_vocabulary
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[ 73%] Built target test_stella_vslam_data_bow_vocabulary
[ 74%] Building CXX object src/pangolin_viewer/CMakeFiles/pangolin_viewer.dir/color_scheme.cc.o
[ 75%] Linking CXX executable test_stella_vslam_data_common_get_cell_indices
[ 75%] Built target test_stella_vslam_data_common_get_cell_indices
[ 76%] Building CXX object test/CMakeFiles/test_stella_vslam_solve_pnp_solver.dir/stella_vslam/solve/pnp_solver.cc.o
[ 77%] Building CXX object test/CMakeFiles/test_stella_vslam_util_fancy_index.dir/stella_vslam/util/fancy_index.cc.o
[ 77%] Linking CXX executable test_stella_vslam_solve_essential_solver
[ 77%] Built target test_stella_vslam_solve_essential_solver
[ 78%] Building CXX object test/CMakeFiles/test_stella_vslam_solve_homography_solver.dir/stella_vslam/solve/homography_solver.cc.o
[ 78%] Linking CXX executable test_stella_vslam_util_fancy_index
[ 78%] Built target test_stella_vslam_util_fancy_index
[ 79%] Building CXX object test/CMakeFiles/test_stella_vslam_util_random_array.dir/stella_vslam/util/random_array.cc.o
[ 80%] Linking CXX executable test_stella_vslam_feature_orb_extractor
[ 81%] Linking CXX executable test_stella_vslam_solve_pnp_solver
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[ 81%] Built target test_stella_vslam_feature_orb_extractor
[ 81%] Built target test_stella_vslam_solve_pnp_solver
[ 82%] Building CXX object test/CMakeFiles/test_stella_vslam_util_trigonometric.dir/stella_vslam/util/trigonometric.cc.o
[ 83%] Linking CXX executable test_stella_vslam_util_random_array
[ 83%] Built target test_stella_vslam_util_random_array
[ 84%] Linking CXX executable test_stella_vslam_solve_fundamental_solver
[ 84%] Built target test_stella_vslam_solve_fundamental_solver
[ 85%] Linking CXX executable test_stella_vslam_util_trigonometric
[ 85%] Built target test_stella_vslam_util_trigonometric
[ 85%] Linking CXX executable test_stella_vslam_solve_homography_solver
[ 85%] Built target test_stella_vslam_solve_homography_solver
[ 86%] Linking CXX shared library ../../lib/libpangolin_viewer.so
[ 86%] Built target pangolin_viewer
Scanning dependencies of target run_video_slam
Scanning dependencies of target run_loop_closure
Scanning dependencies of target run_camera_slam
Scanning dependencies of target run_euroc_slam
Scanning dependencies of target run_tum_rgbd_slam
Scanning dependencies of target run_image_slam
[ 87%] Building CXX object example/CMakeFiles/run_camera_slam.dir/run_camera_slam.cc.o
[ 89%] Building CXX object example/CMakeFiles/run_video_slam.dir/run_video_slam.cc.o
[ 89%] Building CXX object example/CMakeFiles/run_image_slam.dir/run_image_slam.cc.o
[ 89%] Building CXX object example/CMakeFiles/run_euroc_slam.dir/run_euroc_slam.cc.o
[ 90%] Building CXX object example/CMakeFiles/run_loop_closure.dir/run_loop_closure.cc.o
[ 91%] Building CXX object example/CMakeFiles/run_tum_rgbd_slam.dir/run_tum_rgbd_slam.cc.o
[ 92%] Linking CXX executable ../run_camera_slam
[ 92%] Building CXX object example/CMakeFiles/run_image_slam.dir/util/image_util.cc.o
[ 93%] Linking CXX executable ../run_loop_closure
[ 94%] Linking CXX executable ../run_image_slam
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[ 94%] Built target run_camera_slam
[ 94%] Built target run_loop_closure
[ 95%] Building CXX object example/CMakeFiles/run_tum_rgbd_slam.dir/util/tum_rgbd_util.cc.o
Scanning dependencies of target run_kitti_slam
[ 96%] Building CXX object example/CMakeFiles/run_kitti_slam.dir/run_kitti_slam.cc.o
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[ 96%] Built target run_image_slam
[ 97%] Building CXX object example/CMakeFiles/run_euroc_slam.dir/util/euroc_util.cc.o
[ 98%] Building CXX object example/CMakeFiles/run_kitti_slam.dir/util/kitti_util.cc.o
[ 98%] Linking CXX executable ../run_tum_rgbd_slam
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[ 98%] Built target run_tum_rgbd_slam
[ 99%] Linking CXX executable ../run_video_slam
[100%] Linking CXX executable ../run_euroc_slam
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[100%] Built target run_video_slam
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[100%] Built target run_euroc_slam
[100%] Linking CXX executable ../run_kitti_slam
/usr/bin/ld: warning: libopencv_core.so.406, needed by /usr/local/lib/libfbow.so, may conflict with libopencv_core.so.3.4
[100%] Built target run_kitti_slam

**Comment by ymd-stella** [comment, marked as answer]
Same as https://github.com/stella-cv/stella_vslam_ros/discussions/112.
